code/datums/elements/weapon_description.dm

```diff
@@ -0,0 +1,186 @@
+ * examined warning label
+ *
+ * Arguments:
+ *  * tag_val: The value of the item to be added to the tag
+ */
+/datum/element/weapon_description/proc/weapon_tag_convert(tag_val)
+	switch(tag_val)
+		if(0)
+			return "INCAPABLE"
+		if(1 to 25)
+			return "BARELY CAPABLE"
+		if(26 to 50)
+			return "CAPABLE"
+		if(51 to 75)
+			return "VERY CAPABLE"
+		if(76 to INFINITY)
+			return "EXTREMELY CAPABLE"
+		else
+			return "STRANGELY CAPABLE"
+
+
+/**
+ *
+ * Outputs type-specific weapon stats for ballistic weaponry.
+ * It contains extra breaks for the sake of presentation
+ *
+ * Arguments:
+ * 	* source - The object being evaluated
+ */
+/datum/element/weapon_description/proc/add_notes_ballistic(obj/item/gun/ballistic/source)
+	if(source.magazine) // Make sure you have a magazine, thats where the warning is!
+		return "\nBe especially careful around this device, as it can be loaded with <span class='warning'>[source.magazine.caliber]</span> rounds, which you can inspect for more information."
+	else
+		return "\nThe warning attached to the magazine is missing..."
+
+/**
+ *
+ * Outputs type-specific weapon stats for energy-based firearms
+ *
+ * Arguments:
+ * 	* source - The object being evaluated
+ */
+/datum/element/weapon_description/proc/add_notes_energy(obj/item/gun/energy/source)
+	var/list/readout = list("")
+	// Make sure there is something to actually retrieve
+	if(!source.ammo_type)
+		return
+	var/obj/projectile/exam_proj
+	readout += "Standard models of this dangerous tool have <span class='warning'>[source.ammo_type.len]</span> mode(s)"
+	readout += "Our heroic interns have shown that one can theoretically stay standing after..."
+	for(var/obj/item/ammo_casing/energy/for_ammo in source.ammo_type)
+		exam_proj = for_ammo.loaded_projectile
+		if(exam_proj.damage > 0) // Don't divide by 0!!!!!
+			readout += "<span class='warning'>[round(100 / exam_proj.damage, 0.1)]</span> shots on <span class='warning'>[for_ammo.select_name]</span> mode before collapsing from [exam_proj.damage_type == STAMINA ? "immense pain" : "their wounds"]."
+		else
+			readout += "an infinite number of shots on <span class='warning'>[for_ammo.select_name] mode</span>."
+	return readout.Join("\n") // Sending over the singular string, rather than the whole list
+
+/**
+ *
+ * Outputs type-specific weapon stats for ammunition
+ *
+ * Arguments:
+ * 	* source - The object being evaluated
+ */
+/datum/element/weapon_description/proc/add_notes_ammo(obj/item/ammo_casing/source)
+	// Make sure there is actually something IN the casing
+	if(source.loaded_projectile)
+		// No dividing by 0
+		if(source.loaded_projectile.damage > 0)
+			return "Most monkeys our legal team subjected to these rounds succumbed to their wounds after <span class='warning'>[round(100 / (source.loaded_projectile.damage * source.pellets), 0.1)]</span> point-blank discharges, taking <span class='warning'>[source.pellets]</span> shots per round"
```

Comment:
```suggestion
			return "Most monkeys our legal team subjected to these rounds succumbed to their wounds after <span class='warning'>[round(100 / (source.loaded_projectile.damage * source.pellets), 0.1)]</span> point-blank discharges, taking <span class='warning'>[source.pellets]</span> shot\s per round"
```

"point-blank discharges" should also become "point-blank discharge" at 1, but there is no `\es` so you'll need to make a variable.